storm-client/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -781,7 +781,7 @@ private static SerializationDelegate getSerializationDelegate(Map<String, Object
             delegate = (SerializationDelegate) delegateClass.newInstance();
         } catch (ClassNotFoundException | InstantiationException | IllegalAccessException e) {
             LOG.error("Failed to construct serialization delegate, falling back to default", e);
```

Comment:
I would be okay with just turning this into a throw. The default serialization delegate is the GzipThriftSerializationDelegate which we should always be able to instantiate, and I don't see why we wouldn't want to error out if the user tries to use a different delegate and it doesn't work.

Reply:
+1 I would be also OK to not failing back and crash to let users know they set the class incorrectly.